examples/camera/camera_orbit.rs

```diff
@@ -1,22 +1,49 @@
 //! Shows how to orbit camera around a static scene using pitch, yaw, and roll.
+//!
+//! See also: `first_person_view_model` example, which does something similar but as a first-person
+//! camera view.
 
 use std::{f32::consts::FRAC_PI_2, ops::Range};
 
-use bevy::prelude::*;
+use bevy::{
+    input::mouse::{AccumulatedMouseMotion, MouseButtonInput},
+    prelude::*,
+};
 
-#[derive(Debug, Default, Resource)]
+#[derive(Debug, Resource)]
 struct CameraSettings {
     pub orbit_distance: f32,
-    // Multiply keyboard inputs by this factor
-    pub orbit_speed: f32,
+    pub pitch_speed: f32,
     // Clamp pitch to this range
     pub pitch_range: Range<f32>,
+    pub roll_speed: f32,
+    pub yaw_speed: f32,
+}
+
+impl Default for CameraSettings {
+    fn default() -> Self {
+        let pitch_limit = FRAC_PI_2 - 0.01;
+        Self {
+            orbit_distance: 20.0,
+            pitch_speed: 0.003,
+            pitch_range: -pitch_limit..pitch_limit,
+            roll_speed: 0.03,
+            yaw_speed: 0.004,
+        }
+    }
+}
+
+#[derive(Debug, Default, Resource)]
+struct MouseButtonsPressed {
```

Comment:
Maybe the example would be a bit slimmer without this resource? I get that actions should be separated from keycodes, but we read them in the same system as we do the movement right now anyways.

Reply:
This could be me wildly misunderstanding the problem, but: I think only one event gets sent per button pressed (and one on button release). So, the problem becomes how to detect if the user is _holding_ the button, hence the resource.

Reply:
That would be `input.just_pressed`. `input_pressed` is `true` when it was pressed in the first frame and stays true when held the next frames.

Reply:
Edit: Ah, I see your problem! You're using `MouseButtonInput`. Use [`Res<ButtonIntput<MouseButton>>`](https://docs.rs/bevy/latest/bevy/input/struct.ButtonInput.html#method.pressed) instead, that already handles this stuff for you :)

Reply:
Dammit lol I knew something weird was going on!